src/wp-includes/option.php

```diff
@@ -1133,6 +1142,94 @@ function delete_option( $option ) {
 	return false;
 }
 
+/**
+ *  Determines the appropriate autoload value for an option based on input.
+ *
+ *  This function checks the provided autoload value and returns a standardized
+ *  value ('yes', 'no', 'auto-yes', or 'auto-no') based on specific conditions.
+ *
+ * If no valid autoload value is provided, the function will check the size of the given option value
+ * and compare it to the maximum allowed size. If it ls larger, it will return 'auto-no', otherwise
+ * it will return 'auto-yes'.
+ *
+ * @since 6.5.0
+ * @access private
+ *
+ * @param string           $option   The name of the option.
+ * @param mixed            $value    The value of the option to check its autoload value.
+ * @param string|bool|null $autoload The autoload value to check.
+ *                                   Accepts 'yes'|true to enable or 'no'|false to disable, or
+ *                                   'auto-yes' and 'auto-no' for internal purposes.
+ *                                   Any other autoload value will be forced to either `auto-yes`
+ *                                   or `auto-no`.
+ * @return string Returns the original $autoload value if valid, or 'auto-no' if the size exceeds
+ *                the maximum allowed size, or 'auto-yes' otherwise.
+ */
+function determine_option_autoload_value( $option, $value, $autoload ) {
+
+	/**
+	 * Filters the maximum size of option value in bytes.
+	 *
+	 * @param string|bool|null $autoload autoload value to check.
+	 * @param string $option name of the option.
+	 * @param string $value value of the option to check its autoload value.
+	 *
+	 * @return  string|bool|null $autoload autoload value to set. Accepted strings 'yes','no',auto-yes','auto-no' to enable
+	 *
+	 * @since 6.5.0
+	 */
+	$autoload = apply_filters( 'option_autoload_value', $autoload, $option, $value );
+
+	//check if autoload is a boolean.
+	if ( is_bool( $autoload ) ) {
+		return $autoload ? 'on' : 'off';
+	}
+
+	switch ( $autoload ) {
+		case 'yes':
+			return 'on';
+		case 'no':
+			return 'off';
+	}
+
+	/**
+	 * Filter: set the default value if null is passed as the $autoload
+	 *
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param $delfaut  null  The default autoload value.
+	 * @param $autoload string|bool|null The passed autoload value.
+	 * @param $option   string The passed option name.
+	 * @param $value    mixed The passed value to be saved.
```

Comment:
The type and variable names are reversed here. Also, the first parameter needs a bit more explanation, and the second parameter isn't useful to include.
```suggestion
	 * @param bool|null $default The default autoload value to set. Returning true will be set as 'auto-on' in the
	 *                           database, false will be set as 'auto-off', and null will be set as 'auto'.
	 * @param string    $option  The passed option name.
	 * @param mixed     $value   The passed option value to be saved.
```